cpp/include/libxtreemfs/file_info.h

```diff
@@ -211,9 +184,40 @@ class FileInfo {
       bool* wait_completed,
       boost::mutex* wait_completed_mutex);
 
-  /** Get the mutex, that can be used for xLocSet renewals from multiple FileHandles. */
-  inline boost::mutex& xlocset_renewal_mutex() {
-    return xlocset_renewal_mutex_;
+
+  void UpdateXLocSetAndRest(const xtreemfs::pbrpc::XLocSet& new_xlocset,
+                                   bool replicate_on_close);
+
+  void UpdateXLocSetAndRest(const xtreemfs::pbrpc::XLocSet& new_xlocset);
+
+  /** Copies the XlocSet into new_xlocset. */
+  void GetXLocSet(xtreemfs::pbrpc::XLocSet* new_xlocset);
+
+  /** Copies the XlocSet into new_xlocset
+   *  and returns the corresponding UUIDContainer.
+   *  The UUIDcontainer is just valid for the associated XLocSet.
+   */
+  boost::shared_ptr<UUIDContainer> GetXLocSetAndUUIDContainer(
```

Comment:
@noma This is the function is wasn't sure about. Is it bad practice to return something by value and modify a pointer given as a paramter at the same time?